projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/CreateJob.tsx

```diff
@@ -103,7 +112,7 @@ export default class CreateJobDialog extends Component<ICreateJobDialogProps> {
         if (!value) {
             value = this.props.jobPath;
         }
-        sessionStorage.setItem("create-job-name", value);
+        createJobData.jobName = value;
```

Comment:
here, there is `jobName` being a property of an `createJobData`,
then further there is `jobName` being a string key ```input_data["jobName"]```;

So, "jobName" is a part of the data contract with the user. Each user will provide a custom "jobName" value, that lives in context for a particular amount of time (in this case, during a unique `vdk create job` conversation is done).

Maybe we could have the "jobName", and the rest of the data contract, in one place altogether? So the data contract with the user is visible with one glance at the code file :)

Reply:
_**input_data**_ lives in the server extension not in the front-end part, while createJobData, runJobData, etc. live in the front-end
The data entered by the user lives in the front-end until a request is sent to the server where according to the data a specific operation is done. When the operation is done the server sends a specific response to the front-end, after the from-end gets the response all the current saved data is deleted (all the properties are set to "").

About this: **_"Maybe we could have the "jobName", and the rest of the data contract, in one place altogether? So the data contract with the user is visible with one glance at the code file"_** 

- Are you talking about having the data separated into runJobDat, createJobDat, etc.? Maybe it is not good.
- Another solution would be to implement a class: for example UserInputData. And to create a global variable that would "live" from starting the Jupyter Kernel until it stops. The class would have different properties as jobName, jobTeam, etc. and we would only use the needed ones for the specific operation.  In this way it would be easy to see all the information that we would retrieve from the user in one place.

Reply:
I want to simplify it. 
Should be able to link in the VEP -> a file, pointing to a data model stored by VDK. For example, a property we store is `jobName`, and this is described "The name of the data job. Unique workload name given by the a team member.". 

Therefore, in all scopes (session, conversations, requests, etc), `jobName` means that same thing, and I can click in the VEP and navigate to a file, that contains all of those mappings property<->description.